packages/color-scale-block/src/ColorScaleBlock.tsx

```diff
@@ -0,0 +1,539 @@
+import { DndProvider } from 'react-dnd';
+import { HTML5Backend } from 'react-dnd-html5-backend';
+import 'tailwindcss/tailwind.css';
+import { DropZone } from './dragAndDrop/DropZone';
+import { EmptyView } from './components/EmptyView';
+import {
+    calculateDefaultColorWidth,
+    calculateWidths,
+    canExpandColorBlock,
+    defaultColorSquareWidth,
+    resizeEvenly,
+} from './helpers';
+import { MouseEventHandler } from 'react';
+
+type Props = {
+    appBridge: AppBridgeBlock;
+};
+
+export const ColorScaleBlock: FC<Props> = ({ appBridge }) => {
+    const { colorPalettes } = useColorPalettes(appBridge);
+    const [colorPickerPalette, setColorPickerPalette]: [ColorPalette[], (color: ColorPalette[]) => void] = useState(
+        [] as ColorPalette[]
+    );
+    const isEditing = useEditorState(appBridge);
+    const [blockSettings, setBlockSettings] = useBlockSettings<Settings>(appBridge);
+    const [colorScaleHeight, setColorScaleHeight] = useState(
+        blockSettings['customHeight'] ? blockSettings['heightInput'] : blockSettings['heightSlider']
+    );
+    const emptyBlockColors = ['#D5D6D6', '#DFDFDF', '#E8E9E9', '#F1F1F1', '#FAFAFA', '#FFFFFF'];
+    const [currentlyDraggedColorId, setCurrentlyDraggedColorId]: [
+        number | null | undefined,
+        (color?: number | null | undefined) => void
+    ] = useState();
+    const [editedColor, setEditedColor]: [
+        ColorProps | null | undefined,
+        (color: ColorProps | null | undefined) => void
+    ] = useState<Nullable<ColorProps>>();
+    const [isColorPickerOpen, setIsColorPickerOpen] = useState(false);
+    const colorScaleBlockRef: { current: HTMLDivElement | null } = useRef(null);
+    const draggingId: { current?: number | null | undefined } = useRef(null);
+    const dragStartPos: { current?: number | null | undefined } = useRef();
+    const dragStartWidth: { current?: number | null | undefined } = useRef();
+    const lastDragPos: { current?: number | null | undefined } = useRef();
+    const timerToUpdateBlockSettings: { current?: number | undefined } = useRef(undefined);
+    const minimumNumberOfColors = 1;
+    const maximumNumberOfPlaceholderSquares = 6;
+
+    const deleteColor = (id: number | undefined) => {
+        if (!id) {
+            return;
+        }
+
+        const reorderedList = displayableItems?.filter((item: ColorProps) => item.id !== id);
+        setBlockSettings({ ...blockSettings, 'color-input': reorderedList });
+    };
+
+    const calculateLeftPosition = (index: number, width?: number) => {
+        let leftPos = 0;
+        const defaultWidth = width ? width : defaultColorSquareWidth;
+        displayableItems?.map((color: ColorProps, loopIndex: number) => {
+            if (loopIndex < index) {
+                leftPos += color && color.width ? color.width : defaultWidth;
+            }
+            return color;
+        });
+        return leftPos;
+    };
+
+    const updateColor = (newColor: ColorProps) => {
```

Comment:
this can be simplified.

sth like this.?

```
    const updateColor = (newColor: ColorProps) => {
        if (displayableItems.findIndex((item) => item.id === newColor.id) > -1) {
            return;
        }
        const filteredItems = displayableItems.filter((item) => item.color);
        const updatedColors = [...filteredItems, newColor];

        setBlockSettings({
            ...blockSettings,
            'color-input': calculateWidths(updatedColors, colorScaleBlockRef, filteredItems.length > 0),
        });
    };
```

Reply:
There's some logic in calculateWidths that depends on whether this is the first color being added (hence the true/false as the last argument in calculateWidths()).

Let's review this during our call today.